gravitee-apim-console-webui/src/management/api/policy-studio/debug/components/policy-studio-debug-response/policy-studio-debug-response.component.ts

```diff
@@ -137,23 +136,67 @@ export class PolicyStudioDebugResponseComponent implements OnChanges {
           executionTime: debugStep.duration,
           executionStatus: debugStep.status,
           policyName: policy?.name ?? debugStep.policyId,
-          mode: 'POLICY' as const,
+          mode: 'POLICY_RESPONSE' as const,
           icon: policy?.icon,
           flowName: policyScopeToDisplayableLabel(debugStep.scope),
           id: debugStep.id,
+          selected: false,
         };
       }),
       {
         id: 'responseOutput',
         mode: 'RESPONSE_OUTPUT',
+        selected: false,
       },
       {
         id: 'responseClientApp',
         mode: 'CLIENT_APP',
+        selected: false,
       },
     ];
     return timelineSteps;
   }
+
+  private onSelectPolicy(timelineStep: TimelineStep) {
+    const steps = [...this.debugResponse.requestDebugSteps, ...this.debugResponse.responseDebugSteps];
+
+    const outputIndex = steps.findIndex((value) => value.id === timelineStep.id);
+
+    this.inspectorVM = {
+      input:
+        outputIndex > 0
+          ? steps[outputIndex - 1]
+          : {
+              id: 'preprocess-step',
+              policyInstanceId: '',
+              policyId: '',
+              scope: 'ON_REQUEST',
+              status: 'COMPLETED',
+              duration: 0,
+              policyOutput: {
+                ...this.debugResponse.preprocessorStep,
+              },
+            },
+      output: steps[outputIndex],
+    };
```

Comment:
Need to rework this because it will not work if you have selected the first Response policy I think

Reply:
yes and there is missing information. It should also be done earlier I think. In the service ? 

I think we could do a `requestInputDebugStep` like request response DebugStep but for input request. same for output. and use it here for first debug step and for ` Display Request Input - Output`